### ISSUE TITLE
read_ascii is confusing me

### ISSUE THREAD

**Issue by bruceravel** [xraypy/xraylarch#160]
Using [this file](https://raw.githubusercontent.com/bruceravel/demeter/master/t/nonuniform.chi), I do this in Larch:

```
a=read_ascii('t/nonuniform.chi')
show a
== Group ascii_file t/nonuniform.chi: 0 methods, 8 attributes ==
  array_labels: ['example', 'of', ... 'uniform', 'k_grid']
  attrs: <Group header attributes from t/nonuniform.chi>
  column_labels: ['example', 'of', ... 'uniform', 'k_grid']
  data: array<shape=(2, 477), type=dtype('float64')>
  example: array<shape=(477,), type=dtype('float64')>
  filename: 't/nonuniform.chi'
  header: ['# example of chi(k) data that is not on a uniform k-grid']
  of: array<shape=(477,), type=dtype('float64')>
```

The column labels seem to be read from the line *before* the line of dashes.  

To make this more befuddling to me, the example from [section 5.1](http://cars.uchicago.edu/xraylarch/data/index.html#simple-ascii-column-files) of the Larch manual works as expected (by me, at least), with the column labels being taken from the line *after* the line of dashes.

(Using the new simple_labels argument in Athena seems really attractive right now :smile:)

I see that @newville has been mucking around with `read_ascii` in the last few days (when he **should** be enjoying family and the holidays and not working on Larch!!!).  `read_ascii` is a bit complicated, so I am not really prepared to offer a solution right now.  I thought I would raise an issue in case it is something that @newville can fix easily.

**Comment by newville** [comment]
@bruceravel  Yes, I've been playing around with this, trying to be slightly more forgiving, and to try to support some datetime-like columns.    And, yes, during "family quiet study time".

But, your file (and all files in examples/xafsdata) read OK for me (on master branch), with sensibly chosen `array_labels`.  I'll investigate and try to break it...

The docs definitely need work!

**Comment by bruceravel** [comment]
@newville Some more observations:

Firstly, some versioning information:

```
bruce@titanium [demeter](master) > larch
========================================================================
Larch 0.9.32 (2016-Dec-20) M. Newville and Larch Development Team
Using Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609],
Numpy 1.11.3, scipy 0.18.1, matplotlib 1.5.1, and wxpython 3.0.2.0
========================================================================
```

Some background:  Embarrassingly, I realized last week that Demeter was not passing its entire test suite using Larch.  The specific problem here involves Demeter applying heuristics to try to determine if a file contains mu(E,eV), mu(E,keV), or chi(k).

While the header is being read wrongly, the bigger problem is that the data table is read wrongly.

When read as is, the `nonuniform.chi` file does this:

```
larch> show a.data[0,0:5]
array([  1.48341960e+09,   1.28647425e+00,   1.30872275e+00,
         1.34140376e+00,   1.36275577e+00])
larch> show a.data[1,0:5]
array([  1.48341960e+09,   3.14799338e-01,   3.14151878e-01,
         3.13742739e-01,   3.12768783e-01])
larch> show a.data.shape
(2, 477)
```
Yikes!  Those O(1e9) numbers are the source of my problem.

Removing the header entirely yields the correct data table:
```
larch> show a.data[0,0:5]
array([ 1.28647425,  1.30872275,  1.34140376,  1.36275577,  1.3828297 ])
larch> show a.data[1,0:5]
array([ 0.31479934,  0.31415188,  0.31374274,  0.31276878,  0.31159308])
larch> show a.data.shape
(2, 476)
```
I find that if I remove the line of dashes, the data table will be read correctly, although the array labels are not.

With this as the header:
```
# example of chi(k) data that is not on a uniform k-grid
#   k                  chi
```
The data table is correct, but 
```
larch> show a.array_labels
['example', 'of', 'chi_k_', 'data', 'that', '_is', '_not', 'on', 'a', 'uniform', 'k_grid']
```
Removing the `# example of ...` line yields the correct data table, but
```
larch> show a.array_labels
['col1', 'col2']
```

Are we using different versions of python?  numpy?  Something else?

**Comment by newville** [comment]
@bruceravel  Weird.  Thanks for the exact versions, but I doubt it is related to numpy or python versions -- it's probably a parsing mistake on my part.  Will investigate today.

**Comment by newville** [comment]
@bruceravel  I believe this is now fixed with 26b81bc0a2dd1214ac22bf33fb303150e37c76a8  
The attempt to try to allow date-times as data columns was too permissive,  and would incorrectly think that "k" was a date.  I think this may actually be due to the version of "dateutil", which I was not even testing (matplotlib requires it, so it has to be present).

Will add test of reading all the examples/xafsdata files and verifying the values of "array_labels".

**Comment by bruceravel** [comment]
@newville All better in https://github.com/xraypy/xraylarch/commit/26b81bc0a2dd1214ac22bf33fb303150e37c76a8!  Thanks!
